drivers/scale_host.go

```diff
@@ -88,15 +101,26 @@ func (s *ScaleHostDriver) Execute(conf interface{}, apiClient *client.RancherCli
 		return http.StatusInternalServerError, errors.Wrap(err, "Couldn't unmarshal config")
 	}
 
+	hostTemplate, err := apiClient.HostTemplate.ById(config.HostTemplateID)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("Error in get HostTemplate")
+	}
+
+	if hostTemplate == nil || hostTemplate.Removed != "" {
+		return http.StatusBadRequest, fmt.Errorf("HostTemplate does not exist")
+	}
+
 	action := config.Action
 	amount := config.Amount
 	min := config.Min
 	max := config.Max
 	deleteOption := config.DeleteOption
 
 	hostSelector := make(map[string]string)
-	for key, value = range config.HostSelector {
-		hostSelector[key] = value
+	if config.HostSelector != nil {
```

Comment:
I either misread the code originally or this  has been address. Either way, it makes sense now